visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -0,0 +1,154 @@
+		<div class="font-bold tracking-widest pb-2" v-if="label">
+			{{ label.toUpperCase() }}
+		</div>
+		<base-multiple-value-input
+			v-if="isSingleTag"
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+			display-key="value"
+			value-key="value"
+		/>
+		<base-tags-input
+			v-else
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+		/>
+	</div>
+</template>
+
+<script>
+	export default {
+		props: {
+			label: {
+				type: String,
+				default: '',
+			},
+			tKeyColumn: {
+				type: String,
+				default: 'KEY',
+			},
+			tValueColumn: {
+				type: String,
+				default: 'VALUE',
+			},
+			canReloadOnOpen: {
+				type: Boolean,
+				default: false,
+			},
+			isSingleTag: {
+				type: Boolean,
+				default: false,
+			},
+			placeholder: {
+				type: String,
+				default: '',
+			}
+		},
+		data: () => ({
+			is_filter: true,
+			selectedTags: [],
+			initialized: false,
+			tempTags: [], // base-multiple-value-input does not show selected items if `tags` is empty because data is fetched on open.
+		}),
+		computed:{ 
+			tags() {
+				const items = [];
+
+				if (this.rows.length) {
+					for (let row of this.rows) {
+						const key = row[this.tKeyColumn];
+						const value = row[this.tValueColumn];
+						if (key && value) {
+							items.push({
+								key: key.value,
+								value: value.value,
+							})
+						}
+					}
+				}
+				return items;
+			},
+			firstKey() {
+				// Used for single tag
+				if (this.tags.length) {
+					return this.tags[0].key;
+				} else if (this.tempTags) {
+					return this.tempTags[0].key;
+				}
+				return null;
+			}
+		},
+		keyColumn() {
+			return this.tKeyColumn ? this.tKeyColumn : this.findColumnByTag('keys');
+		},
+		valueColumn() {
+			return this.tValueColumn ? this.tValueColumn : this.findColumnByTag('values');
+		},
+		methods: {
+			onVisualizationInit() {
+				const initial_value = this.getFilterValue("selected_items");
+				const selectedTags = [];
+				const tempTags = [];
+
+				if (initial_value) {
+					const urlTags = initial_value.split('|');
+					for (let item of urlTags) {
+						const splitPair = item.split('_');
+						const key = splitPair.length > 1 ? splitPair[1] : splitPair[0];
+						const value = splitPair[0];
+
+						if (key && value) {
+							if (this.isSingleTag) {
+								selectedTags.push(value);
+								tempTags.push({ key, value })
+							} else {
+								selectedTags.push({ key, value });
+							}
+						}
+					}
+				}
+				this.tempTags = tempTags;
+				this.selectedTags = selectedTags;
+			},
+			handleOpen() {
+				if (!this.initialized || this.canReloadOnOpen) {
+					this.fetchLayerData();
+					this.initialized = true;
+				}
```

Comment:
Why do we want to prevent the `fetching` to happen on every `open`?

Reply:
Usually cve lists are used for tags filter and they are huge, might not be needed to fetch them again, but if that does not make sense I can remove this.

Reply:
I'd say we could remove this and look for optimization opportunities once we see how it behaves in production.